src/core/Database.cpp

```diff
@@ -179,8 +179,8 @@ bool Database::save(QString* error, bool atomic, bool backup)
         }
         return false;
     }
-
-    return save(m_data.filePath, error, atomic, backup);
+    const QFileInfo info(m_data.filePath);
+    return save(info.isSymLink() ? info.symLinkTarget() : m_data.filePath, error, atomic, backup);
```

Comment:
You want to use `QFileInfo::canonicalFilePath()` because it returns a blank string if the file does not exist and is a no-op for actual files. See https://doc.qt.io/qt-5/qfileinfo.html#canonicalFilePath

No need to check if the file is a symlink either.

Reply:
However, you will want to make this change in the other save function since that is called by DatabaseWidget. I actually have a pending branch I did to fix this issue and hadn't pushed it yet.